source/common/ssl/connection_impl.cc

```diff
@@ -251,8 +251,21 @@ std::string ConnectionImpl::subjectPeerCertificate() {
   if (!cert) {
     return "";
   }
-  bssl::UniquePtr<char> buf(X509_NAME_oneline(X509_get_subject_name(cert.get()), nullptr, 0));
-  return std::string(buf.get());
+
+  // Format settings for a single line in RFC 2253 format.
+  constexpr int indent = 0;
+  constexpr unsigned long flags = XN_FLAG_RFC2253;
+
+  bssl::UniquePtr<BIO> buf(BIO_new(BIO_s_mem()));
+  X509_NAME_print_ex(buf.get(), X509_get_subject_name(cert.get()), indent, flags);
+  const uint8_t* buf_data;
+  size_t buf_data_len;
+  int mem_content_rc = BIO_mem_contents(buf.get(), &buf_data, &buf_data_len);
+  // It should be impossible for BIO_mem_contents to fail when called with a BIO
+  // created using BIO_s_mem().
+  UNREFERENCED_PARAMETER(mem_content_rc);
```

Comment:
It's just a macro (in `source/common/common/macros.h`) that does ((void)(X)), the point is just to explicitly document it as unreferenced param.